pandas/io/tests/test_html.py

```diff
@@ -627,11 +626,12 @@ def test_parse_dates_combine(self):
         tm.assert_frame_equal(newdf, res[0])
 
     def test_computer_sales_page(self):
+        # See gh-12506. With invalid parsing errors being swallowed,
+        # this test should not return an empty array because none
+        # of the tables could be properly parsed.
         data = os.path.join(DATA_PATH, 'computer_sales_page.html')
-        with tm.assertRaisesRegexp(CParserError, r"Passed header=\[0,1\] are "
-                                   "too many rows for this multi_index "
-                                   "of columns"):
-            self.read_html(data, header=[0, 1])
+        res = self.read_html(data, header=[0, 1])
```

Comment:
@gfyoung no we don't want to swallow legit errors
this was not to change the user api